### PR TITLE
feat: package gurkenlaeufer

### PR DESCRIPTION
**Pull request by tim-goto**
I want to package an [open source header only library](https://github.com/paulelsner/gurkenlaeufer) that we also use internally and would like to fetch using conan.
---

- [X] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [X] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/gurkenlaeufer/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/paulelsner/gurkenlaeufer"
+    topics = ("testing", "header-only")
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
+
+    @property
+    def _min_cppstd(self):
+        return 14
```
Comment (RubenRBS) [inline recipes/gurkenlaeufer/all/conanfile.py:25]: As far as I can see, the recipe says that it supports c++11? Is there a reason I can't see as to why this is 14?

File: recipes/gurkenlaeufer/all/test_package/test_package.cpp
Comment (RubenRBS) [inline recipes/gurkenlaeufer/all/test_package/test_package.cpp:1]: Is this file necessary? Looks like the template was left in

File: recipes/gurkenlaeufer/all/test_v1_package/parsertests.cpp
Comment (RubenRBS) [inline recipes/gurkenlaeufer/all/test_v1_package/parsertests.cpp:1]: Why is this tested only in Conan v1? Tests should test similar functionality in both (But read my global review regarding test complexity)

**Review by RubenRBS — CHANGES_REQUESTED** [review]
Hi! Thanks for your contribution. We really appreciate that you took the time to contribute this recipe to CCI, and sorry about the long wait, we're working hard after the release of Conan 2 and we're still working thru the backlog of recipes :)

The changes overall look good, but I have a question regarding the min cppstd not being the same as the one in the library's readme, and a worry about the test_package's complexity:

[As per the docs](https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/test_packages.md#minimalist-source-code), the test_packages should be as minimal as possible, just enough to ensure a proper package was created and that linkage is possible, which usually is covered by simply printing the library version to stdout and instantiating some trivial objects. We ask this to relive the strain on our CI systems caused by long test_packages, so it would be great if a simpler test package could be made :)

File: recipes/gurkenlaeufer/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+required_conan_version = ">=1.52.0"
+
+
+class GurkenlaeuferConan(ConanFile):
+    name = "gurkenlaeufer"
+    description = "Simple way to write cucumber tests for your C++ projects"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/paulelsner/gurkenlaeufer"
+    topics = ("testing", "header-only")
```
Comment (gjasny) [inline recipes/gurkenlaeufer/all/conanfile.py:19]: Please add `cucumber` and `gherkin`

**Comment by tim-goto** [comment]
No worries, I understand things take longer after such a huge release. Thanks a lot for your feedback, I integrated your suggestions. Please let me know if there is anything else I should take care of.

**Review by RubenRBS — DISMISSED** [review]
Thank you so much for taking the time to make sure the recipe is as great as it is right now.

The changes look nice, thanks for your patience while we reviewed the rest of the backlog after the v2 release :)

File: recipes/gurkenlaeufer/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    def package(self):
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder,"licenses"))
+        copy(self, "*.h", os.path.join(self.source_folder, "include"), os.path.join(self.package_folder, "include"))
+
+    def package_info(self):
+        # Folders not used for header-only
+        self.cpp_info.bindirs = []
+        self.cpp_info.libdirs = []
+
+        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
```
Comment (prince-chrismc) [inline recipes/gurkenlaeufer/all/conanfile.py:74]: You can remove these lines https://github.com/paulelsner/gurkenlaeufer/blob/master/CMakeLists.txt they are not required from what I can see

Comment (tim-goto) [reply recipes/gurkenlaeufer/all/conanfile.py:74]: You mean the cpp_info lines? I thought they should serve as fallback for other packages consuming this one with the older cmake_find_package[_multi] generators. But I guess since no one should be using those removing them should be fine.

**Comment by prince-chrismc** [comment]
Please do not force push 🙏 GitHub forces us to restart the review which is not fun!

File: recipes/gurkenlaeufer/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rmdir
```
Comment (prince-chrismc) [inline recipes/gurkenlaeufer/all/conanfile.py:4]: There are no patches, so less it more :)

```suggestion
from conan.tools.files import get, copy, rmdir
```

File: recipes/gurkenlaeufer/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
```
Comment (prince-chrismc) [inline recipes/gurkenlaeufer/all/conanfile.py:41]:
```suggestion
    def layout(self):
        basic_layout(self, src_folder="src")
```

File: recipes/gurkenlaeufer/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        rmdir(self, os.path.join(self.source_folder, "gtest"))
+
+    def build(self):
+        apply_conandata_patches(self)
```
Comment (prince-chrismc) [inline recipes/gurkenlaeufer/all/conanfile.py:63]:
```suggestion
        pass
```

**Review by prince-chrismc — COMMENTED** [review]
Super tiny nit

File: recipes/gurkenlaeufer/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+
+class GurkenlaeuferConan(ConanFile):
+    name = "gurkenlaeufer"
+    description = "Simple way to write cucumber tests for your C++ projects"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/paulelsner/gurkenlaeufer"
+    topics = ("testing", "header-only", "cucumber", "gherkin")
+    settings = "os", "arch", "compiler", "build_type"
```
Comment (prince-chrismc) [inline recipes/gurkenlaeufer/all/conanfile.py:20]:
```suggestion
    package_type = "header-library"
    settings = "os", "arch", "compiler", "build_type"
```

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]
